app/src/main/java/com/example/demo/kotlin_practice/classesExamples/SecondConstructor.kt

```diff
@@ -0,0 +1,18 @@
+package com.example.demo.kotlin_practice.classesExamples
+
+//Secondary Constructor
+open class SecondConstructor {
+
+    var data: String = ""
+    var numOfData: Int = 0
+
+    constructor(_data: String) {
```

Comment:
never use _ in parameter name,
generally _ is used as prefix for private properties.

Also in you current implementation, named arguments would look weird.

Reply:
Just use the same name as property name and use `this`